### ISSUE TITLE
Update gh-pages branch automatically

### ISSUE THREAD

**Issue by WtfJoke** [codecentric/chaos-monkey-spring-boot#115]
`mvn clean generate-resources site -DperformRelease` <-- updates gh-pages branch automatically

It needs following credentials (id=github) in settings.xml
```
	<server>
		<id>github</id>
		<username>aUserName</username>
		<password>aPasswordOrPersonalAccessToken</password>
	</server>
```
Created @ChaosMonkeyServant for that purpose

Probably its best to change [phase ](https://github.com/codecentric/chaos-monkey-spring-boot/blob/master/chaos-monkey-docs/pom.xml#L98) of site-maven-plugin to deploy and sign everything using (newly created) gpg key (signing is also triggered by `-DperformRelease`.

Successor of #104

**Comment by fletchgqc** [comment]
A related issue: I don't think that we should have documentation for every release version. It's very professional and all, but really it's just a pain. I suggest to just have one set of doco, and if necessary mark features with "since v2.1.0" and stuff. Then less copying around would be needed and it could be easily automated anyway.

**Comment by WtfJoke** [comment]
This was fixed with #204, see https://github.com/codecentric/chaos-monkey-spring-boot/blob/main/.github/workflows/releasetag.yml
